lib/functions.php

```diff
@@ -420,14 +420,26 @@ function deflateResponseFilter(InternalRequest $ireq): \Generator {
         return $headers;
     }
 
-    // Require a text/* mime Content-Type
-    // @TODO Allow option to configure which mime prefixes/types may be compressed
-    if (stripos($headers["content-type"][0], "text/") !== 0) {
-        return $headers;
-    }
-
     $options = $ireq->client->options;
 
+    // Match and cache Content-Type
+    if (!$doDeflate = $options->dynamic_cache->deflateContentTypes[$headers["content-type"][0]] ?? null) {
+        if ($doDeflate === 0) {
+            return $headers;
+        }
+
+        if (count($options->dynamic_cache->deflateContentTypes) == Options::MAX_DEFLATE_ENABLE_CACHE_SIZE) {
+            unset($options->dynamic_cache->deflateContentTypes[key($options->dynamic_cache->deflateContentTypes)]);
+        }
+
+        $doDeflate = preg_match($options->deflateContentTypes, $headers["content-type"][0]);
```

Comment:
This one needs changes. You have to split the `content-type` header, as it may contain additional items, such as the charset.